docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,236 +8,342 @@ title: Authorization
+  providers.  
+- Authorization Servers **MUST** support
+  [OAuth 2.0 Authorization Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414)
+- MCP servers **MUST** respond to an unauthorized request with a `WWW-Authenticate` header containing
+  `issuer`, `scope`, and `resource` identifiers.
+
+The discovery flow is illustrated below:  
 
 ```mermaid
 sequenceDiagram
-    participant B as User-Agent (Browser)
-    participant C as Client
-    participant M as MCP Server
-
-    C->>M: MCP Request
-    M->>C: HTTP 401 Unauthorized
-    Note over C: Generate code_verifier and code_challenge
-    C->>B: Open browser with authorization URL + code_challenge
-    B->>M: GET /authorize
-    Note over M: User logs in and authorizes
-    M->>B: Redirect to callback URL with auth code
-    B->>C: Callback with authorization code
-    C->>M: Token Request with code + code_verifier
-    M->>C: Access Token (+ Refresh Token)
-    C->>M: MCP Request with Access Token
-    Note over C,M: Begin standard MCP message exchange
+    participant MC as MCP Client
+    participant MS as MCP Server
+    participant AUTHZS as Authorization Server
+
+    MC->>MS: Request data
+    MS-->>MC: 401 Unauthorized with WWW-Authenticate
+    MC->>MC: Construct metadata endpoint from issuer identifier
+
+    Note over MC: Continue with authorization flow
 ```
 
-### 2.3 Server Metadata Discovery
+#### 2.3.1 Server Metadata Discovery Headers
+
+MCP clients **SHOULD** include the header `MCP-Protocol-Version: <protocol-version>` during Server
+Metadata Discovery to allow the MCP server to respond based on the MCP protocol version.
+
+For example: `MCP-Protocol-Version: 2024-11-05`
+
+### 2.4 Authorization server integration
+
+#### 2.4.1 Overview
+
+MCP servers **MAY** override endpoints on an as-needed basis if the authorization server requires
+additional gestures that are not captured by the specification.
 
-For server capability discovery:
+MCP servers **MUST NOT** complete PKCE or other public client flows on behalf of the client.
 
-- MCP clients _MUST_ follow the OAuth 2.0 Authorization Server Metadata protocol defined
-  in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
-- MCP server _SHOULD_ follow the OAuth 2.0 Authorization Server Metadata protocol.
-- MCP servers that do not support the OAuth 2.0 Authorization Server Metadata protocol,
-  _MUST_ support fallback URLs.
+MCP servers that **MUST** implement the [metadata discovery process](#23-server-metadata-discovery)
+if they do not use a third-party authorization server.
+
+#### 2.4.2 Flow Description
+
+The third-party authorization flow comprises these steps:
 
-The discovery flow is illustrated below:
+1. MCP client requests data from the MCP server.  
+1. MCP server responds with *HTTP 401 Unauthorized* and `WWW-Authenticate` header containing the
+   `issuer`, `scope`, and `resource`.  
+1. MCP client retrieves the `/authorize` endpoint from the `issuer` metadata document.  
```

Comment:
```suggestion
1. MCP client retrieves the `authorization_endpoint` from the `issuer` metadata document.  
```

Having the slash in this makes it look like the path is hardcoded. The property in the metadata file is `authorization_endpoint`